extensions/amp-story/1.0/amp-story-grid-layer.js

```diff
@@ -115,7 +135,7 @@ export class AmpStoryGridLayer extends AmpStoryBaseLayer {
 
   /** @override */
   prerenderAllowed() {
-    return this.isFirstPage();
+    return this.isFirstPage() || this.isPrerenderActivePage();
```

Comment:
Please factorize these methods so they only run one query.

- if `location.href` has a `page` parameter, run the "am I the initial page" query
- if not, run the "isFirstPage" query